### PR TITLE
docs: proofread market CLI guides

### PR DESCRIPTION
**Pull request by alexandra-mara**
Part of #66, review and proofreading of the Proof Market documentation, [CMD reference](https://docs.nil.foundation/proof-market/market/cmd-reference) section with all its pages.

Includes:
 - [x] proofread for all CLI pages, fixed inaccuracies and updated help messages as they changed a lot since the previous revision
 - [x] added links to the [toolchain](https://github.com/NilFoundation/proof-market-toolchain) repository and explicit mentions of what scripts are described in these pages
 - [x] bid/ask -> request/proposal rename – part of #63
 - [x] proof key updates from #69 and #77, **NOT includes ETH address**
 - [x] private/public option for statements from #81 and #82

TODO:
 - renames like bid/ask pages to request/proposal and CMD to CLI, but those will be added as the next step, as these also require setting up redirects
 - update from #69 and #77 - **ETH address**

### CODE REVIEW

File: gitbook-docs/market/cmd-reference/proof.md [hunk 1]
```diff
@@ -2,60 +2,54 @@
...
+
+usage: proof_tools.py push [-h] [-a request_KEY] [-b proposal_KEY] -f FILE
 
 options:
   -h, --help            show this help message and exit
-  -a ASK_KEY, --ask_key ASK_KEY
-                        ask_key
-  -b BID_KEY, --bid_key BID_KEY
-                        bid_key
+  -a request_KEY, --request_key request_KEY
```
Comment (hgedia) [inline gitbook-docs/market/cmd-reference/proof.md:21]: I think this should have a consistent case. ie REQUEST_KEY. If not, the dev team should update the docs in python.

Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/proof.md:21]: You're right, that's my mistake. Fixed, thank you

**Comment by hgedia** [comment]
LGTM!

**Comment by hgedia** [comment]
LGTM!

File: gitbook-docs/market/cmd-reference/user.md [hunk 2]
```diff
@@ -26,33 +22,32 @@ options:
...
-```
+```json
 {"user":"helloworld","active":true,"extra":{},"error":false,"code":201}
 ```
 
-### **Error#1: Invalid username format**
+#### Error #1: Invalid username format
 
-```
+```json
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/user.md:42]: not json

File: gitbook-docs/market/cmd-reference/README.md [hunk 3]
```diff
@@ -1,3 +1,3 @@
-# CMD Reference
+# CLI Reference
 
-Scripts to interact with the proof market. These are located in `scripts` directory.
+Scripts to interact with the Proof Market, located in the `scripts` directory.
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/README.md:3]:
```suggestion
Scripts for interacting with the Proof Market, located in the `./scripts` directory.
```

File: gitbook-docs/market/cmd-reference/ask.md [hunk 4]
```diff
@@ -1,34 +1,31 @@
 ---
-description: Scripts to manage ask orders
+description: Scripts to manage request orders
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/ask.md:2]:
```suggestion
description: Managing request orders
```

File: gitbook-docs/market/cmd-reference/user.md [hunk 5]
```diff
@@ -1,22 +1,18 @@
 ---
-description: Scripts to manage user
+description: Scripts to manage user authentication
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/user.md:2]:
```suggestion
description: Creating and managing accounts
```

File: gitbook-docs/market/cmd-reference/user.md [hunk 6]
```diff
@@ -1,22 +1,18 @@
...
 ---
 
-# user
+# User
 
-User scripts are used to manage user credentials/accounts
+User scripts are used to manage user credentials/accounts.
 
-## New User
+## New user
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/user.md:9]: Just remove this

Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/user.md:9]:
```suggestion
# Creating a new account```

File: gitbook-docs/market/cmd-reference/user.md [hunk 7]
```diff
@@ -1,22 +1,18 @@
 ---
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/user.md:1]: Headings:

# Creating new accounts
## Usage

python3 signup.py -h
python3 signup.py -u helloworld -p password123 -e hello@test.com

## Success

## Error: bla bla characters

## Error: duplicate username

File: gitbook-docs/market/cmd-reference/user.md [hunk 8]
```diff
@@ -26,33 +24,56 @@ options:
...
+  "_key": "...",
+  "_id": "producer/...",
+  "_rev": "_fneT7J2---",
+  "description": "Generic Producer",
+  "url": "https://example.com",
+  "name": "username",
+  "createdOn": 1677588821180,
+  "updatedOn": 1677588821180
+}
+```
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/user.md:79]: This is a separate issue. https://github.com/NilFoundation/proof-market-toolchain/issues/83

Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/user.md]: Removed this part, saved it in #83

File: gitbook-docs/market/cmd-reference/user.md (hunk 7 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/user.md:1]: Edited this, is the current structure OK?

File: gitbook-docs/market/cmd-reference/user.md (hunk 6 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/user.md]: Fixed

File: gitbook-docs/market/cmd-reference/user.md (hunk 5 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/user.md]: Fixed

File: gitbook-docs/market/cmd-reference/ask.md (hunk 4 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/ask.md]: Fixed

File: gitbook-docs/market/cmd-reference/README.md (hunk 3 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/README.md]: Fixed

File: gitbook-docs/market/cmd-reference/user.md (hunk 2 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/user.md]: Fixed

File: gitbook-docs/market/cmd-reference/user.md (hunk 6 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/user.md]: Is “User account utilities” fine? As this page will later describe not only signing up, but also signing in and registering as a proof producer

File: gitbook-docs/market/cmd-reference/ask.md [hunk 9]
```diff
@@ -1,54 +1,59 @@
...
+    push         push request
+    get          get request
 
+options:
+  -h, --help     show this help message and exit
+  --auth AUTH    auth file
+  -v, --verbose  increase output verbosity
+```
 
+## Submitting proof request
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/ask.md:31]: Submitting a proof request

or

Submitting proof requests

File: gitbook-docs/market/cmd-reference/ask.md [hunk 10]
```diff
@@ -1,54 +1,59 @@
...
 options:
-  -h, --help   show this help message and exit
-  --cost COST  cost
-  --key KEY    key of the statement
-
+  -h, --help            show this help message and exit
+  --cost COST           cost
+  --file FILE           json file with public input
+  --key KEY             statement key
+  --generation_time GENERATION_TIME
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/ask.md:45]: Next thing, we might update these help strings

File: gitbook-docs/market/cmd-reference/README.md [hunk 11]
```diff
@@ -1,3 +1,5 @@
-# CMD Reference
+# CLI Reference
 
-Scripts to interact with the proof market. These are located in `scripts` directory.
+`=nil;` prepared a [toolchain](https://github.com/NilFoundation/proof-market-toolchain)
+for interacting with the Proof Market.
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/README.md:4]: There's something in this phrase that I don't like. The first thing is about preparing. We don't prepare it, we develop it. Next, there could be a more clear call to action.

Use the scripts in the [Proof Market toolchain]() to interact with the Proof Market:

* [Make a new account]()
* [Push your statement as a circuit developer]
* [Order proofs for a zkApp]
* [Propose to sell proofs]
* [Upload and download proofs]

File: gitbook-docs/market/cmd-reference/ask.md [hunk 12]
```diff
@@ -61,27 +66,28 @@ Output
...
 ```
 
-## Retrieve Ask
+## Retrieving info on proof request
+### Usage
 
-```
-python3 ask_tools.py get --h
-usage: ask_tools.py get [-h] [--key KEY]
+Proof requesters can check their requests with the following:
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/ask.md:72]:
```suggestion
Proof requesters can check their requests with the following command:
```

File: gitbook-docs/market/cmd-reference/ask.md [hunk 13]
```diff
@@ -61,27 +66,28 @@ Output
...
+ 
+```console
+foo@bar:~$ python3 request_tools.py get -h
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/ask.md:75]: The `foo@bar` part actually doesn't provide information. Yep, there will be some users on some machines. That's always so.

It's enough to have `$` or `#` in the beginning of a line to have the `console` formatting.

Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/ask.md:75]: But I like that you've joined each example's input and output parts. It just looks better this way.

File: gitbook-docs/market/cmd-reference/ask.md [hunk 14]
```diff
@@ -1,54 +1,59 @@
...
 
-Usage
+`=nil;` prepared a [toolchain](https://github.com/NilFoundation/proof-market-toolchain)
+for interacting with the Proof Market.
+This page describes the script with utilities for proof requesters — `request_tools.py`.
 
-```
-python3 ask_tools.py --h    
-usage: ask_tools.py [-h] [--auth AUTH] {push,get} ...
+Make sure you've [signed in or signed up](user.md) before continuing.
```
[the remark this answers is not in the document]
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/ask.md:11]: Ok. This reminder should go to the User page, right?

File: gitbook-docs/market/cmd-reference/ask.md (hunk 10 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/ask.md:45]: You mean make a task of updating help messages for these scripts?

File: gitbook-docs/market/cmd-reference/ask.md (hunk 13 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/ask.md]: Fixed

File: gitbook-docs/market/cmd-reference/ask.md (hunk 12 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/ask.md]: Fixed

File: gitbook-docs/market/cmd-reference/ask.md (hunk 9 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/ask.md]: Fixed

File: gitbook-docs/market/cmd-reference/README.md [hunk 15]
```diff
@@ -1,3 +1,11 @@
...
+# CLI Reference
 
-Scripts to interact with the proof market. These are located in `scripts` directory.
+Use the scripts from the [Proof Market toolchain](https://github.com/NilFoundation/proof-market-toolchain) to interact with the Proof Market:
+
+* [make a new account](user.md);
+* [push your statement as a circuit developer](statement.md);
+* [order proofs for a zkApp](ask.md);
+* [propose to sell proofs](bid.md);
+* [upload and download proofs](proof.md);
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/README.md:9]: dot in the end?

File: gitbook-docs/market/cmd-reference/statement.md [hunk 16]
```diff
@@ -1,75 +1,104 @@
...
 usage: statement_tools.py push [-h] --file FILE
 
 options:
   -h, --help   show this help message and exit
   --file FILE  file
-
 ```
 
-#### Ex:
+You will be returned an object containing a *_key* field — a unique descriptor of the statement.
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:73]: How exactly will the user receive this object? I don't see it in the console output.

File: gitbook-docs/market/cmd-reference/user.md (hunk 7 above)
Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/user.md:1]: Looks good.

File: gitbook-docs/market/cmd-reference/statement.md [hunk 17]
```diff
@@ -1,75 +1,104 @@
...
+This page describes scripts for circuit developers and proof producers,
+`prepare_statement.py` and `statement_tools.py`.
 
+Make sure you've [signed up](user.md) before continuing.
 
-### Prepare Statement
+## Preparing circuit as a statement
 
-Statements/circuits generated by the zkLLVM compiler need to have additional metadata attached to them before they can be published on the proof market.
+Circuits are stored as a *statement* structure on the Proof Market.
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:14]: The structure is something internal. Okay, it's some data structure inside. But to operate it, we don't need to know it.


```suggestion
Circuits are stored in the form of statements on the Proof Market.
```

File: gitbook-docs/market/cmd-reference/statement.md [hunk 18]
```diff
@@ -1,75 +1,104 @@
 ---
-description: Scripts to manage statements/circuits
+description: Managing circuits and statements
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:2]:
```suggestion
description: Managing statements
```

Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/statement.md:2]: we don't manage circuits on the PM

File: gitbook-docs/market/cmd-reference/statement.md [hunk 19]
```diff
@@ -1,75 +1,104 @@
...
+Make sure you've [signed up](user.md) before continuing.
 
-### Prepare Statement
+## Preparing circuit as a statement
 
-Statements/circuits generated by the zkLLVM compiler need to have additional metadata attached to them before they can be published on the proof market.
+Circuits are stored as a *statement* structure on the Proof Market.
+Statement examples can be found in `./example/statements/` directory.
+Attach all additional metadata to circuits generated by the zkLLVM compiler
+before publishing them on the Proof Market.
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:17]: Looks like this should be in the Examples section

File: gitbook-docs/market/cmd-reference/statement.md [hunk 20]
```diff
@@ -1,75 +1,104 @@
...
+  -t statement_type, --type statement_type
+                        Statement type: placeholder-zkllvm or placeholder-vanilla
+  --private             Make a statement private: only accessible by ID
+  --public              Make a statement public: visible on the Proof Market
 ```
 
-Ex:
+Please note that `--private` and `--public` parameters are mutually exclusive:
+* With `--private`, the statement will be accessible only by its ID.
+  For experiments and development purposes, make your statements private.
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:44]:
```suggestion
  Use private statements for testing circuits in development and making experiments.
```

Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/statement.md:44]: I know that I wrote the original line, but still :)

File: gitbook-docs/market/cmd-reference/statement.md [hunk 21]
```diff
@@ -1,75 +1,104 @@
...
-python3 prepare_statement.py -c arithmetics_example.ll -n arithmetic -t placeholder-zkllvm -o arithmetic.json
+### Example
+
+```bash
+python3 prepare_statement.py -c arithmetics_example.ll -n arithmetic -t placeholder-zkllvm -o \
+arithmetic.json --private
 ```
 
-### Publish Statement
+## Publishing statement
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:54]:
```suggestion
## Publishing statements
```

File: gitbook-docs/market/cmd-reference/statement.md [hunk 22]
```diff
@@ -1,75 +1,104 @@
...
 
-Statements/Circuits can be published to the proof market once modified into the required format. See [prepare statement above](statement.md#prepare-statement).
+Statements/circuits can be published to the Proof Market once modified into the required format.
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:56]: Circuits only appear when we make a statement.

```suggestion
Statements can be published to the Proof Market once modified into the required format.
```

Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/statement.md:56]: Hmm, let's remove and rewrite it.

File: gitbook-docs/market/cmd-reference/statement.md [hunk 23]
```diff
@@ -1,75 +1,104 @@
...
-# statement
+# Statement utilities
 
+This page describes scripts for circuit developers and proof producers,
+`prepare_statement.py` and `statement_tools.py`.
 
+Make sure you've [signed up](user.md) before continuing.
 
-### Prepare Statement
+## Preparing circuit as a statement
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:12]:
```suggestion
## Building statements from circuits
```

File: gitbook-docs/market/cmd-reference/statement.md (hunk 22 above)
Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/statement.md:56]: like that
```suggestion
To publish a statement on the Proof Market, use `statement_tools.py`.
```

File: gitbook-docs/market/cmd-reference/statement.md [hunk 24]
```diff
@@ -1,75 +1,104 @@
...
+## Publishing statement
 
-Statements/Circuits can be published to the proof market once modified into the required format. See [prepare statement above](statement.md#prepare-statement).
+Statements/circuits can be published to the Proof Market once modified into the required format.
+See the [preparing statement](statement.md#preparing-circuit-as-a-statement) section above.
 
-#### Usage
+### Usage
+
+To publish a statement, you'll need to pass its JSON statement description in a file:
```
Comment (NickVolynkin) [inline gitbook-docs/market/cmd-reference/statement.md:61]: no-no, there's no JSON description of a statement. A JSON is the statement itself. The file. But it doesn't matter that it's JSON.

File: gitbook-docs/market/cmd-reference/statement.md (hunk 16 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md:73]: This is a line from our [README](https://github.com/nilfoundation/proof-market-toolchain#2-circuit-generationpublishing-for-circuit-developers). Shall I remove it from here?

File: gitbook-docs/market/cmd-reference/README.md (hunk 15 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/README.md:9]: Thanks, fixed

File: gitbook-docs/market/cmd-reference/statement.md (hunk 18 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md:2]: Thanks, fixed

File: gitbook-docs/market/cmd-reference/statement.md (hunk 17 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md]: Thanks, fixed

File: gitbook-docs/market/cmd-reference/statement.md (hunk 19 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md]: Thanks, moved it

File: gitbook-docs/market/cmd-reference/statement.md (hunk 24 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md]: Removed this for now

File: gitbook-docs/market/cmd-reference/statement.md (hunk 20 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md:43]: Okay)

File: gitbook-docs/market/cmd-reference/statement.md (hunk 23 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md:12]: Thanks, fixed

File: gitbook-docs/market/cmd-reference/statement.md (hunk 16 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md:70]: I see in your examples the following output:
> Statement from /opt/zkllvm-template/build/template.json was pushed with id 12345678
So the id is returned, only not in object. Shall I refrase it like this then?
```suggestion
If successful, you will be returned an id of the statement.
```

Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/statement.md:70]: Use the text from template's readme: https://github.com/NilFoundation/zkllvm-template#step-5-publish-the-circuit-statement

File: gitbook-docs/market/cmd-reference/user.md (hunk 6 above)
Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/user.md]: Yep, it's a good title

Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/user.md]: or just “Account utilities”

File: gitbook-docs/market/cmd-reference/ask.md (hunk 10 above)
Comment (NickVolynkin) [reply gitbook-docs/market/cmd-reference/ask.md:43]: yes, along with more examples and such

File: gitbook-docs/market/cmd-reference/statement.md (hunk 16 above)
Comment (alexandra-mara) [reply gitbook-docs/market/cmd-reference/statement.md]: Updated

**Review by NickVolynkin — APPROVED** [review]

**Pull request merged**